src/jon_mikkel_ex/ex06/logistic_regression.py

```diff
@@ -0,0 +1,333 @@
+# -*- coding: utf-8 -*-
+
+__author__ = "Jon-Mikkel Korsvik"
+__email__ = "jonkors@nmbu.no"
+
+import numpy as np
+from numpy.linalg import norm
+from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.exceptions import NotFittedError
+from sklearn.utils import check_random_state, check_X_y
+
+
+def sigmoid(z):
+    r"""Perform a logistic transform on the input.
+
+    This function applies the sigmoidal function element-wise to all
+    elements of `z`. The sigmoidal function is on the following form:
+
+    .. math::
+
+        \frac{1}{1 + exp(-\mathbf{z})}.
+
+    Parameters
+    ----------
+    z : np.ndarray
+        Logit to transform.
+
+    Returns
+    -------
+    sigmoidal_transformed_z : np.ndarray
+        Transformed input.
+    """
+    if type(z) is int or float:
+        return 1 / (1 + np.exp(-z))
+    if type(z) is np.ndarray:
+        sigmoidal_transformed_z = np.zeros_like(z, np.float64)
+        for ind, val in enumerate(z):
+            sigmoidal_transformed_z[ind] = 1 / (1 + np.exp(-val))
+        return sigmoidal_transformed_z
+    else:
+        raise TypeError('Z input has to be a np.ndarray, alternatively\
+ a float or an int will suffice')
```

Comment:
This trick here is that numpy arrays can be added together just like numbers. All operations are then performed element-wise. A valid solution for this function would be `return 1/(1+np.exp(-z))`. This should work with any np.ndarray.

Reply:
Except of course special operations such as `@`, `.` etc.